src/directory_listing.rs

```diff
@@ -241,7 +246,7 @@ async fn read_dir_entries(
                 }
             };
             if symlink_meta.is_dir() {
-                name_encoded.push('/');
+                uri.push('/');
```

Comment:
There is an additional issue here: we append that slash here but later we don’t consider these symlinks directories (because we only look at `meta.is_dir()` to determine type). Maybe instead of pushing this slash we could set `r#type` here:

```rust
let r#type = if meta.is_dir() {
  dirs_count += 1;
  FileType::DIRECTORY
} else if meta.is_file() {
  …
};
```

Then we can percent-encode the name, append it to the base URI and only then add the slash for directories.

Additional benefit: you won’t have to assume that file names cannot contain slashes. `file\name.txt` is actually a valid file name on Linux, and it needs to be encoded as `file%5Cname.txt` in the URL. Keeping a literal backslash will break the URL.